service/src/main/java/tds/config/services/impl/ConfigServiceImpl.java

```diff
@@ -36,4 +41,46 @@ public ConfigServiceImpl(ConfigRepository configRepository) {
             .filter(f -> f.getAuditObject().equals(type))
             .findFirst();
     }
+
+    /**
+     * This method is meant to replace CommonDLL._FormatMessage_SP (line 1975)
+     * <p>
+     * Instead of following the existing pattern in the code of getting the _key (int) from tds_coremessageobject for default translation
+     * or getting the _key (varbinary) from client_messagetranslation and then using that to get the actual message text.
+     * The repository takes care of that in a single database query.
+     * <p>
+     * NOTE: This method differs from _FormatMessage_SP in that it does not take in arguments to replace placeholder text in the messages.
+     * The consumer of the endpoint is expected to handle this.
+     */
+    @Override
+    @Cacheable(CacheType.LONG_TERM)
+    public String getSystemMessage(String clientName, String messageKey, String language, String context, String subject, String grade) {
+        String clientDefaultLanguage = null;
+        Optional<ClientLanguage> maybeClientLanguage = configRepository.findClientLanguage(clientName);
```

Comment:
I think the solution is to create a method on a different new service or cache the repository call but I didn't want to add caching to another layer on the first pass yet.

Since Spring wraps this service in order to add caching code above the call to the actual method, if you call another method from within the service, even if it has `@Cacheable` it will not go through the caching logic since that is in the wrapper code.